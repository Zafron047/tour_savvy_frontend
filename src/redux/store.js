import { configureStore } from '@reduxjs/toolkit';
import packagesSlice from './packages/packagesSlice';
import reservationsReducer from './reservations/reservationSlice';
import userSlice from './authentication/userSlice';

const loadStateFromLocalStorage = () => {
  try {
    const serializedState = localStorage.getItem('reduxState');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

const loadUserFromLocalStorage = () => {
  try {
    const serializedUser = localStorage.getItem('user');
    if (serializedUser === null) {
      return undefined;
    }
    return JSON.parse(serializedUser);
  } catch (err) {
    return undefined;
  }
};

const store = configureStore({
  reducer: {
    packages: packagesSlice,
    reservations: reservationsReducer,
    current_user: userSlice,
  },
  preloadedState: {
    current_user: loadUserFromLocalStorage(),
    ...loadStateFromLocalStorage(),
  },
  reservations: reservationsReducer,
});

export default store;
