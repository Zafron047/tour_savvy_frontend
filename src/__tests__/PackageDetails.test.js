import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import PackageDetails from '../components/Package_Details';

test('Should render PackageDetails component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <PackageDetails />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
