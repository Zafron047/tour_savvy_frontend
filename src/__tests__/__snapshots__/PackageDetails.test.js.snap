// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PackageDetails component correctly 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
